src/node/interfaces.cpp

```diff
@@ -487,6 +487,14 @@ class ChainImpl : public Chain
         const CChain& active = Assert(m_node.chainman)->ActiveChain();
         return active.GetLocator();
     }
+    CBlockLocator getLocator(const uint256& block_hash) override
+    {
+        LOCK(cs_main);
+        const CChain& active = Assert(m_node.chainman)->ActiveChain();
+        const CBlockIndex *index = m_node.chainman->m_blockman.LookupBlockIndex(block_hash);
+
+        return active.GetLocator(index);
```

Comment:
I kept in `src/node/interfaces.cpp` the same behavior defined in `src/chain.cpp:CCain::GetLocator`.

As this method is only called by `ScanForWalletTransactions()` it will not receive any non-existent blocks. This call only exists inside the condition (`chain().findBlock()` and `if (!block.IsNull())`).

I can change this if necessary.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/25036/commits/37f192b435b03c742aa5f065d1d2d537c4002c94.